src/spikeinterface/sorters/launcher.py

```diff
@@ -4,69 +4,197 @@
 import os
 import stat
 import subprocess
 import sys
+import warnings
 
-from spikeinterface.core import load_extractor, aggregate_units
-from spikeinterface.core.core_tools import check_json
+from spikeinterface.core import aggregate_units
 
 from .sorterlist import sorter_dict
-from .runsorter import run_sorter, run_sorter
-
-
-def _run_one(arg_list):
-    # the multiprocessing python module force to have one unique tuple argument
-    (
-        sorter_name,
-        recording,
-        output_folder,
-        verbose,
-        sorter_params,
-        docker_image,
-        singularity_image,
-        with_output,
-    ) = arg_list
-
-    if isinstance(recording, dict):
-        recording = load_extractor(recording)
+from .runsorter import run_sorter
+from .basesorter import is_log_ok
+
+_default_engine_kwargs = dict(
+    loop=dict(),
+    joblib=dict(n_jobs=-1, backend="loky"),
+    processpoolexecutor=dict(max_workers=2, mp_context=None),
+    dask=dict(client=None),
+    slurm=dict(tmp_script_folder=None, cpus_per_task=1, mem="1G"),
+)
+
+
+_implemented_engine = list(_default_engine_kwargs.keys())
+
+
+def run_sorter_jobs(job_list, engine="loop", engine_kwargs={}, return_output=False):
+    """
+    Run several :py:func:`run_sorter()` sequentially or in parallel given a list of jobs.
+
+    For **engine="loop"** this is equivalent to:
+
+    ..code::
+
+        for job in job_list:
+            run_sorter(**job)
+
+    The following engines block the I/O:
+      * "loop"
+      * "joblib"
+      * "multiprocessing"
+      * "dask"
+
+    The following engines are *asynchronous*:
+      * "slurm"
+
+    Where *blocking* means that this function is blocking until the results are returned.
+    This is in opposition to *asynchronous*, where the function returns `None` almost immediately (aka non-blocking),
+    but the results must be retrieved by hand when jobs are finished. No mechanisim is provided here to be aware
+    when jobs are finish.
+    In this *asynchronous* case, the :py:func:read_sorter_folder() helps to retrieve individual results.
+
+
+    Parameters
+    ----------
+    job_list: list of dict
+        A list a dict that are propagated to run_sorter(...)
+    engine: str "loop", "joblib", "dask", "slurm"
+        The engine to run the list.
+        * "loop": a simple loop. This engine is
+    engine_kwargs: dict
+
+    return_output: bool, dfault False
+        Return a sorting or None.
+
+    Returns
+    -------
+    sortings: None or list of sorting
+        With engine="loop" or "joblib" you can optional get directly the list of sorting result if return_output=True.
+    """
+
+    assert engine in _implemented_engine, f"engine must be in {_implemented_engine}"
+
+    engine_kwargs_ = dict()
+    engine_kwargs_.update(_default_engine_kwargs[engine])
+    engine_kwargs_.update(engine_kwargs)
+    engine_kwargs = engine_kwargs_
+
+    if return_output:
+        assert engine in ("loop", "joblib", "processpoolexecutor")
```

Comment:
Assertion message missing
```suggestion
        assert engine in ("loop", "joblib", "processpoolexecutor"), "Only 'loop', 'joblib', and 'processpoolexecutor' support return_output=True."
```